packages/core/lib/modules/wallet.js

```diff
@@ -0,0 +1,231 @@
+import {v4 as uuid} from 'uuid';
+import {DockRpc} from '../client/dock-rpc';
+import {KeyringRpc} from '../client/keyring-rpc';
+import {UtilCryptoRpc} from '../client/util-crypto-rpc';
+import {WalletRpc} from '../client/wallet-rpc';
+import {getRealm, initRealm} from '../core/realm';
+import {DocumentType, WalletDocument} from '../types';
+import {EventManager} from './event-manager';
+import {NetworkManager} from './network-manager';
+import {Accounts} from './accounts';
+
+export const NetworkConfigs = {
+  mainnet: {
+    name: 'Dock PoS Mainnet',
+    url: 'wss://mainnet-node.dock.io',
+    addressPrefix: 22,
+  },
+  testnet: {
+    name: 'Dock PoS Testnet',
+    url: 'wss://knox-1.dock.io',
+    addressPrefix: 21,
+  },
+  local: {
+    name: 'Local Node',
+    url: 'ws://127.0.0.1:9944',
+    addressPrefix: 21,
+  },
+};
+
+export const WalletEvents = {
+  ready: 'ready',
+  documentAdded: 'document-added',
+  documentUpdated: 'document-updated',
+  documentRemoved: 'document-removed',
+  networkUpdated: 'network-updated',
+  networkConnected: 'network-connected',
+};
+
+export type WalletStatus = 'closed' | 'loading' | 'ready';
+
+/** Wallet */
+export class Wallet {
+  networkManager: NetworkManager;
+  context: string[];
+  status: WalletStatus;
+  apiConnected: boolean;
+  eventManager: EventManager;
+  walletId: string;
+  accounts: Accounts;
+
+  constructor({
+    walletId = 'dock-wallet',
+    context = ['https://w3id.org/wallet/v1'],
+  } = {}) {
+    this.walletId = walletId;
+    this.context = context;
+    this.networkManager = NetworkManager.getInstance();
+    this.eventManager = new EventManager();
+    this.status = 'closed';
+    this.eventManager.registerEvents(WalletEvents);
+    this.accounts = Accounts.getInstance({wallet: this});
+  }
+
+  async close() {
+    getRealm().close();
```

Comment:
any chance of errors when closing one of these that could crash the app in an unsafe way (e.g. a file handle is left open or something? I'm not sure how the mobile OS's handle crashing processes that have files still open)

Reply:
it's handled by realm, it should open the db without issues.